src/scenes/Engagement/Files/useUploadEngagementFiles.ts

```diff
@@ -0,0 +1,54 @@
+import { GQLOperations } from '../../../api';
+import {
+  HandleUploadCompletedFunction,
+  UploadFilesConsumerFunction,
+  UploadFilesConsumerInput,
+  useUploadFiles,
+} from '../../../components/files/hooks';
+import {
+  useUploadInternshipEngagementGrowthPlanMutation,
+  useUploadLanguageEngagementPnpMutation,
+} from './UploadEngagementFiles.generated';
+
+type EngagementType = 'language' | 'internship';
+
+export const useUploadEngagementFile = (
+  engagementType: EngagementType
+): UploadFilesConsumerFunction => {
+  const uploadFiles = useUploadFiles();
+
+  const [uploadPnp] = useUploadLanguageEngagementPnpMutation();
+  const [uploadGrowthPlan] = useUploadInternshipEngagementGrowthPlanMutation();
+
+  const uploadFile =
+    engagementType === 'language' ? uploadPnp : uploadGrowthPlan;
+
+  const handleUploadCompleted: HandleUploadCompletedFunction = async ({
+    uploadId,
+    name,
+    parentId: id,
+    action,
+  }) => {
+    await uploadFile({
+      variables: {
+        id,
+        pnp: { uploadId, name },
+        growthPlan: { uploadId, name },
```

Comment:
Does Apollo/GQL not complain about unused variables here?

Reply:
Not so far! I initially did something different that didn't involve passing these, but I switched to this because it was cleaner, and I didn't get any warnings or errors in the console.